### ISSUE TITLE
Reading large file and slow processing each line lead to Stream read timeout

### ISSUE THREAD

**Issue by ganuonglachanh** [sharpart555/nexline#22]
I'm reading large file (10 MB->1GB) and then processing each line with slow function, then at the end of file (last line), the **nl.next()** fire a timeout error ( `can't catch this error`):
```
UnhandledPromiseRejectionWarning: Error: Stream read timeout!
    at Timeout.setTimeout [as _onTimeout] (somedrive:\somewhere\node_modules\nexline\reader\streamReader.js:31:13)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
```

```
while (true) {
        const line = await nl.next().catch((err) => { return null }); // -> error here and can't catch this error
        if (line === null) {
            break; // If all data is read, returns null
        }
        await slowfunction(line);
}
```

**Comment by sharpart555** [comment]
Hi, thanks for reporting!

Can you give me more information about this issue?
* Did you use `stream` for `nexline` input?
* Could you try `file descriptor` rather than `stream` for `nexline` input?

**Comment by ganuonglachanh** [comment]
Hi, I’m using **stream** for input

**Comment by sharpart555** [comment]
Handle stream errors inside `nexline` is not good way to solve this problem
Stream emits error as event, but `nexline` is not `EventEmitter`.
To convert stream error event to throwable error, `nexline` must add event listener to `stream`
but adding an event listener to an external input can cause side effect.
I think proper way to handle stream error is listen error event from stream object directly.

Or, you can use `file descriptor` rather than `stream` which is not `EventEmitter` and has no `timeout`.

**Comment by sharpart555** [comment]
OK I got it.
I removed stream read timeout in nexline@1.1.1.
You can install from now.
Please let me know if your problem is solved.

**Comment by sharpart555** [comment]
I think this issue is fixed at nexline@1.1.1.
If your problem is not solved, feel free to reopen it.
